python-threatexchange/threatexchange/cli/hash_cmd.py

```diff
@@ -88,21 +89,7 @@ def __init__(
         self.files = files
 
     def execute(self, settings: CLISettings) -> None:
-        hashers = [
-            s
-            for s in settings.get_signal_types_for_content(self.content_type)
-            if issubclass(s, FileHasher)
-        ]
-        if self.signal_type is not None:
-            if self.signal_type not in hashers:
-                raise CommandError.user(
-                    f"{self.signal_type.get_name()} "
-                    f"does not apply to {self.content_type.get_name()}"
-                )
-            hashers = [self.signal_type]  # type: ignore  # can't detect intersection types
-
-        for file in self.files:
-            for hasher in hashers:
-                hash_str = hasher.hash_from_file(file)
-                if hash_str:
-                    print(hasher.get_name(), hash_str)
+        hasher = Hasher(settings, self.content_type, self.signal_type)
```

Comment:
blocking: Unfortunately, anything that is being passed in the settings god object is still heavily tied in with the CLI.